src/overlays/actors/ovl_kaleido_scope/z_kaleido_equipment.c

```diff
@@ -1,6 +1,595 @@
+                if ((cursorItem >= ITEM_BULLET_BAG_30) && (cursorItem <= ITEM_BULLET_BAG_50)) {
+                    pauseCtx->nameColorSet = 1;
+                } else {
+                    pauseCtx->nameColorSet = 0;
+                }
+            } else {
+                pauseCtx->nameColorSet = 0;
+            }
+        }
+
+        KaleidoScope_SetCursorVtx(pauseCtx, cursorSlot * 4, pauseCtx->equipVtx);
+
+        if ((pauseCtx->cursorSpecialPos == 0) && (cursorItem != PAUSE_ITEM_NONE) && (pauseCtx->state == 6) &&
+            (pauseCtx->unk_1E4 == 0) && CHECK_BTN_ALL(input->press.button, BTN_A) &&
+            (pauseCtx->cursorX[PAUSE_EQUIP] != 0)) {
+
+            if ((gEquipAgeReqs[pauseCtx->cursorY[PAUSE_EQUIP]][pauseCtx->cursorX[PAUSE_EQUIP]] == 9) ||
+                (gEquipAgeReqs[pauseCtx->cursorY[PAUSE_EQUIP]][pauseCtx->cursorX[PAUSE_EQUIP]] ==
+                 ((void)0, gSaveContext.linkAge))) {
+                Inventory_ChangeEquipment(pauseCtx->cursorY[PAUSE_EQUIP], pauseCtx->cursorX[PAUSE_EQUIP]);
+
+                if (pauseCtx->cursorY[PAUSE_EQUIP] == 0) {
+                    gSaveContext.infTable[29] = 0;
+                    gSaveContext.equips.buttonItems[0] = cursorItem;
+
+                    if ((pauseCtx->cursorX[PAUSE_EQUIP] == 3) && (gSaveContext.bgsFlag != 0)) {
+                        gSaveContext.equips.buttonItems[0] = ITEM_SWORD_BGS;
+                        gSaveContext.swordHealth = 8;
+                    } else {
+                        if (gSaveContext.equips.buttonItems[0] == ITEM_HEART_PIECE_2) {
+                            gSaveContext.equips.buttonItems[0] = ITEM_SWORD_BGS;
+                        }
+                        if ((gSaveContext.equips.buttonItems[0] == ITEM_SWORD_BGS) && (gSaveContext.bgsFlag == 0) &&
+                            (gBitFlags[3] & gSaveContext.inventory.equipment)) {
+                            gSaveContext.equips.buttonItems[0] = ITEM_SWORD_KNIFE;
+                        }
+                    }
+
+                    Interface_LoadItemIcon1(globalCtx, 0);
+                }
+
+                Audio_PlaySoundGeneral(NA_SE_SY_DECIDE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                pauseCtx->unk_1E4 = 7;
+                sEquipTimer = 10;
+            } else {
+                Audio_PlaySoundGeneral(NA_SE_SY_ERROR, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            }
+        }
+
+        if (oldCursorPoint != pauseCtx->cursorPoint[PAUSE_EQUIP]) {
+            Audio_PlaySoundGeneral(NA_SE_SY_CURSOR, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    } else if ((pauseCtx->unk_1E4 == 7) && (pauseCtx->pageIndex == PAUSE_EQUIP)) {
+        KaleidoScope_SetCursorVtx(pauseCtx, pauseCtx->cursorSlot[PAUSE_EQUIP] * 4, pauseCtx->equipVtx);
+        pauseCtx->cursorColorSet = 8;
+
+        sEquipTimer--;
+        if (sEquipTimer == 0) {
+            pauseCtx->unk_1E4 = 0;
+        }
```

Comment:
I guess `DECR` doesn't work here, right?

Reply:
`DECR` wouldn't work here because the decrement is done even if `sEquipTimer` is 0